From e109fcfb252c17cc6956210e68bf99c48af719e6 Mon Sep 17 00:00:00 2001
From: Erik Fleckstein <erik@tinkerforge.com>
Date: Thu, 17 Feb 2022 10:20:07 +0100
Subject: [PATCH] esp_http_server: Don't handle pings in
 httpd_ws_get_frame_type.

The registered web socket handler (if it handles control frames) is
called when a ping frame is received, but if it was already handled
here, httpd_ws_recv_frame will not work in the handler, breaking the
connection.
---
 components/esp_http_server/src/httpd_ws.c | 21 ---------------------
 1 file changed, 21 deletions(-)

diff --git a/components/esp_http_server/src/httpd_ws.c b/components/esp_http_server/src/httpd_ws.c
index 4a4c2373df..b3702ac581 100644
--- a/components/esp_http_server/src/httpd_ws.c
+++ b/components/esp_http_server/src/httpd_ws.c
@@ -463,27 +463,6 @@ esp_err_t httpd_ws_get_frame_type(httpd_req_t *req)
     aux->ws_final = (first_byte & HTTPD_WS_FIN_BIT) != 0;
     aux->ws_type = (first_byte & HTTPD_WS_OPCODE_BITS);
 
-    /* Reply to PING. For PONG and CLOSE, it will be handled elsewhere. */
-    if(aux->ws_type == HTTPD_WS_TYPE_PING) {
-        ESP_LOGD(TAG, LOG_FMT("Got a WS PING frame, Replying PONG..."));
-
-        /* Read the rest of the PING frame, for PONG to reply back. */
-        /* Please refer to RFC6455 Section 5.5.2 for more details */
-        httpd_ws_frame_t frame;
-        uint8_t frame_buf[128] = { 0 };
-        memset(&frame, 0, sizeof(httpd_ws_frame_t));
-        frame.payload = frame_buf;
-
-        if(httpd_ws_recv_frame(req, &frame, 126) != ESP_OK) {
-            ESP_LOGD(TAG, LOG_FMT("Cannot receive the full PING frame"));
-            return ESP_ERR_INVALID_STATE;
-        }
-
-        /* Now turn the frame to PONG */
-        frame.type = HTTPD_WS_TYPE_PONG;
-        return httpd_ws_send_frame(req, &frame);
-    }
-
     return ESP_OK;
 }
 
-- 
2.35.1

